### ISSUE TITLE
waterway=fish_pass

### ISSUE THREAD

**Issue by printmaps** [gravitystorm/openstreetmap-carto#2895]
Fish ladders are currently not rendered. I recommend to draw them identically as waterway=stream.

**Comment by kocio-pl** [comment]
Sounds reasonably:

https://wiki.openstreetmap.org/wiki/Tag%3Awaterway%3Dfish_pass

**Comment by polarbearing** [comment]
[taginfo](https://taginfo.openstreetmap.org/tags/waterway=fish_pass) 719 on a way, 17 on a node.

**Comment by pnorman** [comment]
The usage seems insufficient.

**Comment by kocio-pl** [comment]
What do you think would be sufficient usage in this case?

**Comment by pnorman** [comment]
There isn't a specific number. And if there were more usage, we'd still need to consider if it's desirable to render it, since usage alone doesn't mean we want to display it.

**Comment by kocio-pl** [comment]
It's best to know all such things beforehand to make deciding easier. I don't expect a number or hard limits, even general estimation would be helpful.

**Comment by dieterdreist** [comment]
2017-10-18 3:16 GMT+02:00 Paul Norman <notifications@github.com>:

> There isn't a specific number. And if there were more usage, we'd still
> need to consider if it's desirable to render it, since usage alone doesn't
> mean we want to display it.
>


these are (artificial) parts of natural waterways like rivers, I don't see
why we wouldn't want to render them. 700 seems already quite significant
usage for a relatively rare feature like this.

**Comment by printmaps** [comment]
Please consider that a fishpass divides the landscape and is the reason for bridges. Example (the small waterway is the fishpass):

![bildschirmfoto 2017-10-18 um 11 56 33](https://user-images.githubusercontent.com/26689799/31712517-9e1562a6-b3fb-11e7-86ce-51bfe3e9b969.png)

**Comment by math1985** [comment]
I dont think we should increase the amount of tags in the waterway namespace for such specialised purposes. That will make life difficult for every data consumer. Better something like waterway=canal (or ditch?), canal=fish_pass.

**Comment by dieterdreist** [comment]
2017-10-18 12:08 GMT+02:00 Matthijs Melissen <notifications@github.com>:

> I dont think we should increase the amount of tags in the waterway
> namespace for such specialised purposes. That will make life difficult for
> every data consumer. Better something like waterway=canal (or ditch?),
> canal=fish_pass.
>

yes, waterway classification is a pity. For fish passes, none of the
established artificial waterway tags apply: canal (usually navigable)
doesn't seem appropriate (they're too small), neither are (by their
meaning) ditch (at least many are not ditches) or "drain" (nothing drained
here). You can always subtag the details, but the generic class should
contain the subclasses, otherwise this is tagging for the renderer.

**Comment by kocio-pl** [comment]
It's definitely not a canal:

> Use waterway=canal for man-made waterways used for transportation or also for the largest waterways created for irrigation purposes. 

It has also nothing to do with drainage system:

> Use waterway=drain for minor artificial waterways, typically lined with concrete or similar, used to carry storm water or grey-discharge. 

> Use waterway=ditch for simple narrow artificial waterways used to drain nearby land, to remove storm-water or similar.

I agree that this would be tagging for rendering and we don't know how many of them are tagged this way already. Unfortunately waterway=* also contains other objects than just a water line (like dam or weir), so it's not easy for data consumer to pick just the water, but our database has no explicit "contains water" property.

**Comment by kocio-pl** [comment]
> I dont think we should increase the amount of tags in the waterway namespace for such specialised purposes. That will make life difficult for every data consumer.

Honestly I don't understand - how would rendering already existing tag on osm-carto make problems for data consumer?

**Comment by SomeoneElseOSM** [comment]
> Honestly I don't understand - how would rendering already existing tag on osm-carto make problems for data consumer?

I'm not sure I understand that either, but http://map.atownsend.org.uk/maps/map/map.html does render fish passes (in the UK and Ireland) so if there's a problem, someone should be able to find an example there.

**Comment by ZeiP** [comment]
The usage has grown and there are some places where the tagging has not been used because it's not rendered. In some cases the fish pass is separate from the main waterway, so in my opinion it's important to have the fish passes rendered, and I support the suggested rendering as a stream – this is also mostly used in the cases where the tagging has been done based on the rendering.

**Comment by jeisenbe** [comment]
There is some debate about this proposed tag on https://wiki.openstreetmap.org/wiki/Talk:Tag:waterway%3Dfish_pass - It would be good to discuss further whether `waterway=fish_pass` or `waterway=canal` + `canal=`  or `usage=` =fish_pass` is preferable. Note that `waterway=canal` is now used with `usage=spillway` on dams, a similar feature is some ways.

**Comment by gpsvisualizer** [comment]
Regardless of how the debate turns out on the OSM wiki, is there any reason NOT to simply render waterway=fish_pass as either a canal or stream?

**Comment by jeisenbe** [comment]
If there are a significant number of fish passes which are tagged with `waterway=canal` + `usage=fish_pass` then that would be a problem. At the moment this does not appear to be the case: https://taginfo.openstreetmap.org/search?q=usage%3Dfish_pass (16 uses)

If someone wants to submit a [new PR](https://github.com/gravitystorm/openstreetmap-carto/compare) showing a rendering for this feature, and tests how it works in a number of places, it would be welcomed. @gpsvisualizer would you be interested in doing this?

**Comment by gpsvisualizer** [comment]
I have no idea how to do "pull requests."  I don't even understand what Git is, to be honest — and my brain is full enough already.

But I still don't understand why it'd be a problem if both schemes were rendered, regardless of how many tags exist out there.  What's the harm?

**Comment by SomeoneElseOSM** [comment]
> I have no idea how to do "pull requests."

Just in case you (or someone else) did want to do this, I wrote a diary entry a while ago that was designed to be a step-by-step guide to what would be required, with no prior knowledge of git or github: https://www.openstreetmap.org/user/SomeoneElse/diary/43041 .  I'm sure other guides are available too.

**Comment by sun-geo** [comment]
Small update, the count of `waterway=fish_pass` crosses the 1k line, see

![image](https://user-images.githubusercontent.com/47290585/91204530-870ec900-e704-11ea-92c4-a6354bca02d2.png)
<sup>https://ohsome.org/apps/dashboard</sup>

The current hack for displaying fish passes would be, while having the 1d ![](https://wiki.openstreetmap.org/w/images/thumb/0/0f/Mf_way.svg/20px-Mf_way.svg.png) `waterway=fish_pass` tag by additional adding a 2d ![](https://wiki.openstreetmap.org/w/images/thumb/0/0f/Mf_area.svg/20px-Mf_area.svg.png) water area e.g. using a tagging like `natural=water` + `water=fish_pass` ;-)

See examples [here ](https://sophox.org/embed.html#%23defaultView%3AMap%0ASELECT%20%2a%20WHERE%20%7B%0A%20%20%3FosmId%20osmt%3Anatural%20%22water%22%20%3B%0A%20%20%20%20%20%20%20%20%20osmt%3Awater%20%22fish_pass%22%20%3B%0A%20%20%20%20%20%20%20%20%20osmm%3Aloc%20%3Floc%20.%0A%7D).

**Comment by jeisenbe** [comment]
Marked as “de facto” in the wiki by @matkoniecz and now up to 1300 uses: https://wiki.openstreetmap.org/w/index.php?title=Tag:waterway%3Dfish_pass&curid=107799&diff=2083627&oldid=2060920

**Comment by TheJJ** [comment]
I've naively submitted #4388 to render this just like a stream for now, after noticing that my "de-facto" tagging wouldn't show up :)

Although I'm a total noob, using `waterway=canal` + `usage=fish_pass` seems very good for the connecting waterway, just the fish steps (-> not a canal) itself could maybe better be rendered with `waterway=fish_pass`.
What do you think?

**Comment by BertMule** [comment]
I ran into the same issue.
It is so disappointing this is open since 2017!

Why is this made so difficult?
Just render it as a ditch or something. Done.

These discussions take longer than the actual implementation.
And nothing comes out of it.
Often it is just rejected, or left to die.

I probably have to change it to a ditch myself.

**Comment by ppete2** [comment]
I also would recommend rendering _waterway=fish_pass_ ! Style: Like _waterway=stream_
Not only in OSM, but also in reality lots of fish-passes have been built the last 10-15 years due to new nature conservation/water laws. Opposite a natural stream they are shorter and its main purpose is venting water barriers for fishes and other water creatures.

**Comment by imagico** [comment]
This already has a PR in #4388 that just needs a bit of work to be finalized.  What needs to be done can be found in https://github.com/gravitystorm/openstreetmap-carto/pull/4388#issuecomment-1805843117.

**Comment by ppete2** [comment]
Just for interest: When will fishpasses finally be visible in osm.org - map? 1,5 months since pull request got merged, and not visible yet.

**Comment by dch0ph** [comment]
@ppete2 The OSM website only updates the stylesheet when there is a new release. The frequency of new Carto releases is typically annual.
